### PR TITLE
Improved mean squared error. 

### PR DESCRIPTION
**Pull request by xiejw**
This PR has the following changes.
1. Fixed the docstring for the meanSquaredError. The original version is likely due to copy-paste.
2. Restricted the wrt of gradient to make it consistent with other APIs.
3. Added numerical tests for loss and gradients.

### CODE REVIEW

File: Sources/DeepLearning/Loss.swift [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
 
-/// Computes the mean squared error between logits and labels.
+/// Computes the mean squared error between predictions and labels.
 ///
 /// - Parameters:
-///   - logits: One-hot encoded outputs from a neural network.
-///   - labels: One-hot encoded values that correspond to the correct output.
-@differentiable
+///   - predicted: Predicted outputs from a neural network.
+///   - labels: Expected values, i.e., targets, that correspond to the correct output.
```
Comment (rxwei) [inline Sources/DeepLearning/Loss.swift]:
```suggestion
///   - labels: Expected values, i.e. targets, that correspond to the correct output.
```
